src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Cache.cs

```diff
@@ -254,7 +254,11 @@ private static void Add(Key key, Regex regex)
                 }
 
                 // Finally add the regex.
-                var node = new Node(key, regex);
+                var node = new Node(key, regex)
+                {
+                    LastAccessStamp = s_lastAccessed is { } lastAccessed ? Volatile.Read(ref lastAccessed.LastAccessStamp) + 1 : 1
```

Comment:
If s_lastAccessed is null, so we need to set LastAccessedStamp? Why not just let it be 0 in that case?

Reply:
Yes, it can. Fixed, thanks.